tasks/agent6-linux.yml

```diff
@@ -10,6 +10,13 @@
     dest: /etc/datadog-agent
     state: directory
 
+- name: Create system-probe configuration file
+  template:
+    src: system-probe.yaml.j2
+    dest: /etc/datadog-agent/system-probe.yaml
+    owner: "root"
```

Comment:
should this be owned by dd-agent ? The process-agent needs to read the system-probe.yaml file, so making the owner root could be a problem

Reply:
Its currently owned by root in our environments.

Reply:
I do think it should be owned by root - is there a way to confirm that it'll be readable by dd-agent? (or additional config to change the chmod settings?)

Reply:
I was told it will be own by root as a security requirement. If it's own by dd-agent then we must move it inside the datadog.yaml like every other "datadog" configuration.

Reply:
> I was told it will be own by root as a security requirement. If it's own by dd-agent then we must move it inside the datadog.yaml like every other "datadog" configuration.

what is the security-gain of a root-ownership especially on this file? if your thinking is: "the unpriviledged datadog-user should not be able to modify its own configuration" than *all* config should be owned by root and only readable by the datadog_group.

Reply:
The system-probe needs to run with elevated privileges as it uses eBPF, and so we do not want the unprivileged datadog-user to make changes to a configuration file that a privileged binary reads - though since the unprivileged user still needs to access it, it should have read-only rights.